nixos/modules/services/hardware/pid-fan-controller.nix

```
@@ -0,0 +1,190 @@
+          This path can contain multiple wildcards, but has to resolve to
+          exactly one result.
+        '';
+      };
+      pidParams = {
+        setPoint = lib.mkOption {
+          type = lib.types.ints.unsigned;
+          description = "Set point of the controller in °C";
+        };
+        P = lib.mkOption {
+          description = "K_p of PID controller";
+          type = lib.types.float;
+        };
+        I = lib.mkOption {
+          description = "K_i of PID controller";
+          type = lib.types.float;
+        };
+        D = lib.mkOption {
+          description = "K_d of PID controller";
+          type = lib.types.float;
+        };
+      };
+    };
+  };
+
+  fan = {
+    options = {
+      wildcardPath = lib.mkOption {
+        type = lib.types.str;
+        description = ''
+          Wildcard path of the hwmon 'pwm' file.
+          If the fans are not to be found in /sys/class/hwmon/hwmon* the corresponding
+          kernel module (like 'nct6775') needs to be added to `boot.kernelModules`.
+          See: [hwmon](https://www.kernel.org/doc/html/latest/hwmon/index.html)
+        '';
+      };
+      minPwm = lib.mkOption {
+        default = 0;
+        type = lib.types.ints.u8;
+        description = "Minimum PWM value";
+      };
+      maxPwm = lib.mkOption {
+        default = 255;
+        type = lib.types.ints.u8;
+        description = "Maximum PWM value";
+      };
+      cutoff = lib.mkOption {
+        default = false;
+        type = lib.types.bool;
+        description = "Whether to stop the fan when `minPwm` is reached";
+      };
+      heatPressureSrcs = lib.mkOption {
+        type = lib.types.nonEmptyListOf lib.types.str;
+        description = "Heat pressure sources affected by the fan";
+      };
+    };
+  };
+in
+{
+  options.services.pid-fan-controller = {
+    enable = lib.mkEnableOption ''
+      the PID fan controller, which controls the configured fans by running
+      a closed-loop PID control loop.
+    '';
+    package = lib.mkPackageOption pkgs "pid-fan-controller" { };
+    settings = {
+      interval = lib.mkOption {
+        default = 500;
+        type = lib.types.int;
+        description = "Interval between controller cycles in ms";
+      };
+      heatSrcs = lib.mkOption {
+        type = lib.types.listOf (lib.types.submodule heatSource);
+        description = "List of heat sources to be monitored";
+        example = ''
+          [
+            {
+              name = "cpu";
+              wildcardPath = "/sys/devices/pci0000:00/0000:00:18.3/hwmon/hwmon*/temp1_input";
+              pidParams = {
+                setPoint = 60;
+                P = -5.0e-3;
+                I = -2.0e-3;
+                D = -6.0e-3;
+              };
+            }
+          ];
+        '';
+      };
+      fans = lib.mkOption {
+        type = lib.types.listOf (lib.types.submodule fan);
+        description = "List of fans to be controlled";
```

```suggestion
        description = "List of fans to be controlled by the PID Fan Controller.";
```

Alternatively, just add the period.